scenario_player/utils/reclaim.py

```diff
@@ -0,0 +1,520 @@
+        if node.address not in address_to_proxy_manager:
+            client = JSONRPCClient(web3, node.privkey)
+            address_to_proxy_manager[node.address] = get_proxy_manager(client, deploy)
+        proxy_manager = address_to_proxy_manager[node.address]
+        (userdeposit_proxy, user_token_proxy) = get_udc_and_corresponding_token_from_dependencies(
+            chain_id=chain_id, proxy_manager=proxy_manager
+        )
+
+        balance = userdeposit_proxy.get_total_deposit(to_canonical_address(node.address), "latest")
+        log.debug("UDC balance", balance=balance, address=node.address)
+        if balance > 0:
+            drain_amount = TokenAmount(balance)
+            log.info(
+                "Planning withdraw",
+                from_address=node.address,
+                amount=drain_amount.__format__(",d"),
+            )
+            try:
+                ready_at_block = userdeposit_proxy.plan_withdraw(drain_amount, "latest")
+            except InsufficientEth:
+                log.warning("Not sufficient eth in node wallet to withdraw", address=node.address)
+                continue
+            planned_withdraws[node.address] = ready_at_block, drain_amount
+
+    if not planned_withdraws:
+        return
+
+    log.info(
+        "Withdraws successfully planned, waiting until withdraws are ready",
+        planned_withdraws=planned_withdraws,
+    )
+
+    reclaim_amount = 0
+    for address, (ready_at_block, amount) in planned_withdraws.items():
+        proxy_manager = address_to_proxy_manager[address]
+        (userdeposit_proxy, user_token_proxy) = get_udc_and_corresponding_token_from_dependencies(
+            chain_id=chain_id, proxy_manager=proxy_manager
+        )
+        # FIXME: Something is off with the block numbers, adding 20 to work around.
+        #        See https://github.com/raiden-network/raiden/pull/6091/files#r412234516
+        proxy_manager.client.wait_until_block(BlockNumber(ready_at_block + 20), retry_timeout=10)
+
+        log.info("Withdraw", user_address=address, amount=amount.__format__(",d"))
+        reclaim_amount += amount
+        userdeposit_proxy.withdraw(amount, "latest")
+
+    log.info(
+        "All UDC withdraws finished, now claim the resulting tokens!",
+        withdraw_total=reclaim_amount.__format__(",d"),
+    )
+
+    reclaim_erc20(
+        reclamation_candidates,
+        userdeposit_proxy.token_address("latest"),
+        contract_manager,
+        account,
+        eth_rpc_endpoint,
+    )
+
+
+def reclaim_erc20(
+    reclamation_candidates: List[ReclamationCandidate],
+    token_address: TokenAddress,
+    contract_manager: ContractManager,
+    account: Account,
+    eth_rpc_endpoint: URI,
+):
+    web3 = Web3(HTTPProvider(eth_rpc_endpoint))
+
+    reclaim_amount = 0
+
+    log.info("Checking chain for claimable tokens", token_address=token_address)
+    for node in reclamation_candidates:
+        client = JSONRPCClient(web3, node.privkey)
```

Comment:
Same problem here ... This also has an unnecessary runtime cost to the Ethereum node, since each of these classes perform runtime checks which require REST calls, which may or not be expensive. But as we have experience before, the nodes are not able to handle a lot of concurrent action (specially with logs), so we should avoid these unecessary extra calls.

Additionally, it looks like these objects are being used through out the functions, so it makes sense to have a single utility to instantiate them all.